ambry-router/src/main/java/com/github/ambry/router/GetBlobInfoOperation.java

```diff
@@ -142,20 +143,25 @@ void poll(RequestRegistrationCallback<GetOperation> requestRegistrationCallback)
    * @param requestRegistrationCallback The callback to use to notify the networking layer of dropped requests.
    */
   private void cleanupExpiredInFlightRequests(RequestRegistrationCallback<GetOperation> requestRegistrationCallback) {
-    Iterator<Map.Entry<Integer, GetRequestInfo>> inFlightRequestsIterator =
+    Iterator<Map.Entry<Integer, RequestInfo>> inFlightRequestsIterator =
         correlationIdToGetRequestInfo.entrySet().iterator();
     while (inFlightRequestsIterator.hasNext()) {
-      Map.Entry<Integer, GetRequestInfo> entry = inFlightRequestsIterator.next();
+      Map.Entry<Integer, RequestInfo> entry = inFlightRequestsIterator.next();
       int correlationId = entry.getKey();
-      GetRequestInfo info = entry.getValue();
-      if (time.milliseconds() - info.startTimeMs > routerConfig.routerRequestTimeoutMs) {
+      RequestInfo requestInfo = entry.getValue();
+      // If there is no response from network layer or if request itself didn't get chance to reach chance to reach
+      // network layer (due to some throttling in router, etc), drop the request after some time.
+      long currentTimeInMs = time.milliseconds();
+      if ((requestInfo.isRequestReceivedByNetworkLayer()
```

Comment:
We should abstract this if condition into a separate method (maybe RouterUtils.isRouterRequestExpired()). Its the same condition for all Operations.